src/NHibernate/Linq/GroupBy/GroupBySelectClauseRewriter.cs

```diff
@@ -67,8 +75,8 @@ protected override Expression VisitMember(MemberExpression expression)
 				return base.VisitMember(expression);
 			}
 
-			if ((elementSelector is NewExpression || elementSelector.NodeType == ExpressionType.Convert)
```

Comment:
Instead of checking whether it is a `ExpressionType.Convert` expression, its operand is checked in order to verify whether it is actually supported. The `ExpressionType.Convert` was used for `ExpressionType.ArrayIndex` expression which was added by #433.